.github/workflows/python-test.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will install Python dependencies and run tests with a variety of Python versions and OSes
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python tests
+
+on: push
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
```

Comment:
Let's also specify minimum verified versions in the README. I believe the README calls out 3.10.10 and poetry 1.2.

Let's add "3.11" to the tests as well?